src/inmanta/execute/runtime.py

```diff
@@ -233,15 +317,37 @@ class DelayedResultVariable(ResultVariable[T]):
         (a queue variable can be dequeued by the scheduler when a provider is added)
     """
 
-    __slots__ = ("queued", "queues", "listeners")
+    __slots__ = ("queued", "queues", "listeners", "promises", "done_promises")
 
     def __init__(self, queue: "QueueScheduler", value: Optional[T] = None) -> None:
         ResultVariable.__init__(self, value)
+        self.promises: List[IPromise] = []
```

Comment:
There variables will have to be set to None upon Freeze, to prevent a major memory leak I think.